src/core/annotation.js

```diff
@@ -3540,28 +3541,34 @@ class FreeTextAnnotation extends MarkupAnnotation {
   constructor(params) {
     super(params);
 
-    this.data.hasOwnCanvas = this.data.noRotate;
+    this.data.hasOwnCanvas = true;
 
     const { xref } = params;
     this.data.annotationType = AnnotationType.FREETEXT;
     this.setDefaultAppearance(params);
     if (!this.appearance && this._isOffscreenCanvasSupported) {
       const strokeAlpha = params.dict.get("CA");
       const fakeUnicodeFont = new FakeUnicodeFont(xref, "sans-serif");
-      const fontData = this.data.defaultAppearanceData;
+      const { fontColor, fontSize } = this.data.defaultAppearanceData;
       this.appearance = fakeUnicodeFont.createAppearance(
         this._contents.str,
         this.rectangle,
         this.rotation,
-        fontData.fontSize || 10,
-        fontData.fontColor,
+        fontSize || 10,
+        fontColor,
         strokeAlpha
       );
+      this.data.fontColor = Array.from(fontColor);
+      this.data.fontSize = fontSize || 10;
       this._streams.push(this.appearance, FakeUnicodeFont.toUnicodeStream);
     } else if (!this._isOffscreenCanvasSupported) {
       warn(
         "FreeTextAnnotation: OffscreenCanvas is not supported, annotation may not render correctly."
       );
+    } else if (this.appearance) {
+      const { fontColor, fontSize } = parseAppearanceStream(this.appearance);
+      this.data.fontColor = Array.from(fontColor);
+      this.data.fontSize = fontSize;
     }
```

Comment:
Are all of these `else if` branches actually working as intended now?
Would it not be more correct to reformat [all this code](https://github.com/mozilla/pdf.js/pull/16522/files#diff-a7e409abd3d3209d32d3adc8457c5e23feb4c06113a2e065b0c6a9322a3a3369R3549-R3572) as follows:
```javascript
if (this.appearance) {
  const { fontColor, fontSize } = parseAppearanceStream(this.appearance);
  ...
} else if (this._isOffscreenCanvasSupported) {
  const strokeAlpha = params.dict.get("CA");
  ...
} else {
  warn(
    "FreeTextAnnotation: OffscreenCanvas is not supported, annotation may not render correctly."
  );
```